test/models/media_test.rb

```diff
@@ -1032,4 +1033,14 @@ class HttpRedirectionLoop < StandardError
     assert_equal api_key.settings[:config][:storage], Pender::Store.current.instance_variable_get(:@storage)
   end
 
+  test "should not get metrics from Facebook when facebook api key is empty" do
+    url = 'http://www.facebook.com/people/\u091C\u0941\u0928\u0948\u0926-\u0905\u0939\u092E\u0926/100014835514496'
+    api_key = create_api_key application_settings: { config: { facebook:{ app_id: '', app_secret: '' } }}
+    response = Media.get_metrics_from_facebook(url, api_key.id, 10)
+    assert_nil response
+    m = create_media url: url
+    data = m.as_json
+    assert_nil data['metrics']['facebook']
```

Comment:
If there's a problem with the metrics request, Pender expect it to be an empty hash (`{}`), not `nil`
https://github.com/meedan/pender/blob/develop/app/models/concerns/media_facebook_engagement_metrics.rb#L31-L34